### PR TITLE
z_obj_mure.c

### PR DESCRIPTION
**Pull request by mzxrules**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

I don't have anything clever to say for this one

### CODE REVIEW

File: src/overlays/actors/ovl_Obj_Mure/z_obj_mure.c [hunk 1]
```diff
@@ -31,53 +32,366 @@ const ActorInit Obj_Mure_InitVars = {
...
+    /* 0 */ OBJMURE_TYPE_GRASS,
+    /* 1 */ OBJMURE_TYPE_UNDEFINED,
+    /* 2 */ OBJMURE_TYPE_FISH,
+    /* 3 */ OBJMURE_TYPE_BUGS,
+    /* 4 */ OBJMURE_TYPE_BUTTERFLY
+} ObjMureType;
+
+typedef enum {
+    /* 0 */ OBJMURE_CHILD_STATE_0,
+    /* 1 */ OBJMURE_CHILD_STATE_1, // Dead
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Obj_Mure/z_obj_mure.c:70]: Why not rename it to `OBJMURE_CHILD_STATE_DEAD`?

**Review by AngheloAlf — APPROVED** [review]

File: src/overlays/actors/ovl_Obj_Mure/z_obj_mure.c [hunk 2]
```diff
@@ -31,53 +32,366 @@ const ActorInit Obj_Mure_InitVars = {
...
+} ObjMureType;
+
+typedef enum {
+    /* 0 */ OBJMURE_CHILD_STATE_0,
+    /* 1 */ OBJMURE_CHILD_STATE_DEAD,
+    /* 2 */ OBJMURE_CHILD_STATE_2
+} ObjMureChildState;
+
+s32 func_808D78D0(ObjMure* this, GlobalContext* globalCtx) {
+    if (this->type == 2 || this->type == 3 || this->type == 4) {
```
Comment (tom-overton) [inline src/overlays/actors/ovl_Obj_Mure/z_obj_mure.c:75]: Are these checking against `ObjMureType`? In other words, would this code be correct?
```suggestion
    if (this->type == OBJMURE_TYPE_FISH || this->type == OBJMURE_TYPE_BUGS || this->type == OBJMURE_TYPE_BUTTERFLY) {
```

File: src/overlays/actors/ovl_Obj_Mure/z_obj_mure.c [hunk 3]
```diff
@@ -31,53 +32,366 @@ const ActorInit Obj_Mure_InitVars = {
...
+    s32 maxChildren = ObjMure_GetMaxChildSpawns(this);
+    s32 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure/func_808D7C64.s")
+    for (i = 0; i < maxChildren; i++) {
+        ObjMure_GetSpawnPos(&spawnPos, &actor->world.pos, this->ptn, i);
+        this->children[i] = Actor_SpawnAsChildAndCutscene(
+            &globalCtx->actorCtx, globalCtx, sSpawnActorIds[this->type], spawnPos.x, spawnPos.y, spawnPos.z,
+            actor->world.rot.x, actor->world.rot.y, actor->world.rot.z,
+            (this->type == 4 && i == 0) ? 1 : sSpawnParams[this->type], this->actor.cutscene, this->actor.unk20, NULL);
```
Comment (engineer124) [inline src/overlays/actors/ovl_Obj_Mure/z_obj_mure.c:176]:
```suggestion
            (this->type == OBJMURE_TYPE_BUTTERFLY && i == 0) ? 1 : sSpawnParams[this->type], this->actor.cutscene, this->actor.unk20, NULL);
```

**Review by engineer124 — COMMENTED** [review]
Just building off the previous comment. An extra spot for the type enum

**Review by Kenix3 — APPROVED** [review]

File: src/overlays/actors/ovl_Obj_Mure/z_obj_mure.c (hunk 2 above)
Comment (engineer124) [reply src/overlays/actors/ovl_Obj_Mure/z_obj_mure.c:75]: I'm good to approve once these are set

File: src/overlays/actors/ovl_Obj_Mure/z_obj_mure.c [hunk 4]
```diff
@@ -31,53 +32,367 @@ const ActorInit Obj_Mure_InitVars = {
...
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure/func_808D78D0.s")
+    this->chNum = OBJ_MURE_GET_CHNUM(this->actor.params);
+    this->ptn = OBJ_MURE_GET_PTN(this->actor.params);
+    this->svNum = OBJ_MURE_GET_SVNUM(this->actor.params);
+    this->type = OBJ_MURE_GET_TYPE(this->actor.params);
+    if (this->ptn >= 4) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+    if (this->type >= 5) {
```
Comment (tom-overton) [inline src/overlays/actors/ovl_Obj_Mure/z_obj_mure.c:101]: You could add a `OBJMURE_TYPE_MAX` to the type enum above and then do:
```suggestion
    if (this->type >= OBJMURE_TYPE_MAX) {
```

**Review by tom-overton — APPROVED** [review]

File: src/overlays/actors/ovl_Obj_Mure/z_obj_mure.c (hunk 4 above)
Comment (EllipticEllipsis) [reply src/overlays/actors/ovl_Obj_Mure/z_obj_mure.c:101]: I agree with this. I'll approve and merge once this is done.

**Pull request merged**